### DISCUSSION TITLE
How do I make bat auto-detect the language when the input is piped?

### DISCUSSION THREAD

**Discussion by trajano** [sharkdp/bat#2774] [Q&A]
How do I make bat auto-detect the language when the input is piped, is it even possible?

I tried

bat --language=auto
[bat error]: unknown syntax: 'auto'

There's a `libmagic` perhaps it can be set to wait for 100ms (configurable ideally) to see if it can be determined from the stream before rendering

**Comment by keith-hall** [comment]
When piping into `bat`, `bat` should try to autodetect the language automatically without any special command line flags. However, it is limited to looking at only the first line of input currently. Can you share some example inputs which you expect to work but which don't please?
Also, https://github.com/sharkdp/bat/issues/2699 seems relevant here

**Reply by trajano** [reply]
```
docker ps --format json | bat
```
![image](https://github.com/sharkdp/bat/assets/110627/4e069563-21ed-4a8f-a842-4d1ba1a45cb9)

![image](https://github.com/sharkdp/bat/assets/110627/0b23f196-1426-47cf-aa38-b7db122fda55)

I also tried `command bat` to ensure I'm running `bat` without any aliasing
